lib/output.rb

```diff
@@ -2,106 +2,124 @@
+  CODE_GENERATED = blue_b('The secret code has been generated, you may begin . . .')
+  GAME_BEGIN = green_b('Would you like to begin? [y/n]')
+  GAME_AGAIN = green_b('Would you like to go again? [y/n]')
+  NO_GUESSES = red_b('Nice try, but you ran out of guesses :(')
+  THE_ANSWER = white_b('The secret code was: ')
+  ERRORS = red_b('Error(s):')
+  LENGTH_ERROR = red_b('The input is an incorrect length. The input should be inputing 4 characters')
+  TYPE_ERROR = red_b('That\'s not the correct input type. I need nums plz')
+  CONTEXT_ERROR = red_b('Hmmmmmmmm. I know arrays start at 0, but I want values 1-6 x|')
+
+  def print_intro
+    puts ''
+    header
+    instructions
+    type_ok
+    system 'clear'
+  end
 
-    def print_start_again(first_time_p = false)
-      if first_time_p
-        print GAME_BEGIN
-      else
-        print GAME_AGAIN
-      end
+  def print_start_again(first_time_p = false)
+    if first_time_p
+      print GAME_BEGIN
+    else
+      print GAME_AGAIN
     end
+  end
 
-    def print_input_prompt(guesses_left)
-      puts GUESSES_LEFT + Colors.cyan(guesses_left.to_s)
-      print INPUT_PROMPT
-      return gets.chomp
-    end
+  def print_input_prompt(guesses_left)
+    puts GUESSES_LEFT + cyan(guesses_left.to_s)
+    print INPUT_PROMPT
+    return gets.chomp
+  end
 
-    def print_no_guesses
-      system 'clear'
-      puts NO_GUESSES
-    end
+  def print_no_guesses
+    system 'clear'
+    puts NO_GUESSES
+  end
 
-    def print_the_code
-      puts THE_ANSWER + Colors.yellow_b(@de_code.join(' '))
-    end
+  def print_code_generated
+    puts CODE_GENERATED
+  end
 
-    def print_input_error
-      puts INPUT_ERROR
-    end
+  def print_the_code
+    puts THE_ANSWER + yellow_b(@de_code.join(' '))
+  end
 
-    def print_code_generated
-      puts CODE_GENERATED
-    end
+  def print_error_start
+    puts "\n\t" + ERRORS
+  end
 
-    def gimme_de_code(de_code)
-      @de_code = de_code
-    end
+  def print_input_error
+    puts INPUT_ERROR
+  end
+
+  def print_length_error
+    puts "\n\t\t" + LENGTH_ERROR
+  end
+
+  def print_type_error
+    puts "\n\t\t" + TYPE_ERROR
+  end
+
+  def print_context_error
+    puts "\n\t\t" + CONTEXT_ERROR
+  end
+
+  def gimme_de_code(de_code)
+    @de_code = de_code
+  end
 
     private
 
-      def instructions
-        puts RULES
-      end
-
-      def header
-        puts TITLE_STARS
-        puts TITLE
-        puts TITLE_STARS
-      end
-
-      def type_ok
-        print ("Type 'ok' to continue: ")
-        response = gets.chomp
-        if response.downcase.eql? 'ok'
-          return
-        else
-          type_ok
-        end
-      end
+  def instructions
+    puts RULES
+  end
+
+  def header
+    puts TITLE_STARS
+    puts TITLE
+    puts TITLE_STARS
+  end
+
+  def type_ok
+    print ("Type 'ok' to continue: ")
+    response = gets.chomp
+    if response.downcase.eql? 'ok'
```

Comment:
unless response.downcase == 'ok'
 type_ok
end

Reply:
Right, guard cases; thx for pointing that out.

Is there any reason you prefer == over .eql? or is it simply convention?

Reply:
They're _almost_ always interchangeable, and for your comparison here, using `==` would be more conventional.